lib/api/core/storage/storageEngine.js

```diff
@@ -144,28 +150,70 @@ class IndexCache {
   remove ({ index, collection, scope='public', notify=true }) {
     let modified = false;
 
-    if (this.indexes[index] && this.indexes[index].scope === scope) {
-      if (collection) {
-        const position = this.indexes[index].collections.indexOf(collection);
+    if ( this._indexes[index]
+      && this._indexes[index].scope === scope
+      && collection)
+    {
+      const position = this._indexes[index].collections.indexOf(collection);
 
-        if (position >= 0) {
-          this.indexes[index].collections.splice(position, 1);
-          modified = true;
+      if (position !== -1) {
+        this._indexes[index].collections.splice(position, 1);
+
+        if (this._indexes[index].collections.length === 0) {
```

Comment:
Since indexes are just volatile collection containers, we delete them from the cache when there is no collection left inside